### PR TITLE
Do not require stop_code to be unique

### LINKED ISSUE #987: Stop code uniqueness should not be mandatory
## Observed behavior (please include a screenshot if possible)

When creating a stop, if another stop has the same `stop_code` as the stop being created, the stops editor shows a validation error, claiming that stop codes need to be unique, and does not allow saving the stop.

## Expected behavior

Stop codes should be allowed to be non-unique. _The GTFS specification does not require uniqueness of stop codes_ and there are real-life cases where this is true, for instance, in Ottawa, Canada, OC Transpo assigns one stop code to each Transitway station, which contains multiple bus platforms, each represented as a distinct stop (`location_type = 0`) in GTFS. I.e. Stops A, B, C and D at Tunney's Pasture station [all bear code 3011.](https://www.octranspo.com/en/our-services/stations1/tunneys-pasture1/#:~:text=to%20the%20train!-,Station%20Layout,-Stops%20and%20Service)

## Steps to reproduce the problem

1. In GTFS editor, create a stop and assign it a stop code, then save it.
2. Create a second stop and assign it the same stop code as the stop created in step 1.

## Version of datatools-ui and datatools-server if applicable (exact commit hash or branch name)

Commit `6fc72d2148e8dda005aae9c2b26ba911e4197506 `

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Fix #987 . Stop_code should not be required to be unique. This PR also slightly amends the help text associated with stop_code.

### CODE REVIEW

File: gtfs.yml [hunk 1]
```diff
@@ -147,10 +147,10 @@
     required: true
     inputType: GTFS_ID
     columnWidth: 6
-    helpContent: "The stop_id field contains an ID that uniquely identifies a stop or station. Multiple routes may use the same stop. The stop_id is dataset unique."
+    helpContent: "The stop_id field contains an ID that identifies a stop or station. Multiple routes may use the same stop. The stop_id is dataset unique."
```
Comment (binh-dam-ibigroup) [inline gtfs.yml]: The PR description says the help text for `stop_code`, not `stop_id` was supposed be changed. Which one is correct?

**Review by binh-dam-ibigroup — APPROVED** [review]
Just one small question about the documentation.

File: gtfs.yml (hunk 1 above)
Comment (philip-cline) [reply gtfs.yml]: Good catch, should be stop_code. Updated in https://github.com/ibi-group/datatools-ui/pull/993/commits/e09c8a80627dc0ae8c3a8f28ab4dc6ed7777aa7f

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks good, but can we add a validation item that says it's nice if they are?

**Comment by philip-cline** [comment]
I will ticket adding "nice to have" validation, since that's not currently configurable.

**Pull request merged**